modules/ossm-migrating-from-smcp-defined-gateways-to-gateway-injection.adoc

```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v2x/ossm-gateway-migration.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="ossm-migrating-from-smcp-defined-gateways-to-gateway-injection_{context}"]
+= Migrate from SMCP-Defined gateways to gateway injection
+:context: ossm-migrating-from-smcp-defined-gateways-to-gateway-injection
+
+This procedure explains how to migrate with zero downtime from gateways defined in the `ServiceMeshControlPlane` resource to gateways that are managed using gateway injection. This migration is achieved by using the existing gateway Service resource to target a new gateway deployment that is created using gateway injection.
+
+.Prerequisties
+
+* You are logged in to the {ocp-product-title} web console as `cluster-admin`.
+
+* The {SMProductName} Operator must be installed.
+
+* The `ServiceMeshControlPlane` resource must be deployed and an ingress gateway exists in the configuration.
+
+.Procedure
+
+. Create a new ingress gateway that is configured to use gateway injection. 
++
+[NOTE]
+====
+This procedure migrates away from the default ingress gateway deployment defined in the `ServiceMeshControlPlane` resource to gateway injection. The procedure may be modified to migrate from additional ingress gateways configured in the SMCP.    
+====
++
+[source,yaml, subs="attributes,verbatim"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway-canary 
+  namespace: istio-system <1>
+spec:
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+      istio: ingressgateway 
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels: <2>
+        app: istio-ingressgateway
+        istio: ingressgateway
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto
+      serviceAccountName: istio-ingressgateway
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-reader
+  namespace: istio-system
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: istio-ingressgateway-secret-reader
+  namespace: istio-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secret-reader
+subjects:
+  - kind: ServiceAccount
+    name: istio-ingressgateway
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy <3>
+metadata:
+  name: gatewayingress
+  namespace: istio-system
+spec:
+  podSelector:
+    matchLabels:
+      istio: ingressgateway
+  ingress:
+    - {}
+  policyTypes:
+    - Ingress
+---- 
```

Comment:
It's out of my scope to check for technical accuracy, but is this all meant to be in one big YAML file? Or are these meant to be distinct YAML files that you create separately? If it's the latter, they need to be split out into separate files, and likely their own steps. Something like:

. Create a new ingress gateway that is configured to use gateway injection:

.. Create YAML file 1

.. Create YAML file 2

etc

Ignore this comment if that's not the case, and let me know if you have questions about what I mean here

Reply:
This is one big YAML file. I considered linking to a repo that contains the file, but we decided to put all the code in the topic.